exchange/exchange.go

```diff
@@ -137,31 +137,31 @@ func (ex *Exchange) RemoveQueueBindings(queueName string) []*binding.Binding {
 }
 
 // GetMatchedQueues returns queues matched for message routing key
-func (ex *Exchange) GetMatchedQueues(message *amqp.Message) (matchedQueues map[string]bool) {
+func (ex *Exchange) GetMatchedQueues(message *amqp.Message) (matchedQueues []string) {
 	// @spec-note
 	// The server MUST implement these standard exchange types: fanout, direct.
 	// The server SHOULD implement these standard exchange types: topic, headers.
 
-	// TODO implement "headers" exchange
-	matchedQueues = make(map[string]bool)
+	allQueues := make([]string, 0, 1)
+	matchedQueues = make([]string, 0, 1)
```

Comment:
I'm not sure this will improve performance, maybe for the simple case where only one queue matches, but for multiple matches, this will likely make it worse because of the filtering at the end of the function.

Optimizing matches might be done through a caching system, I feel the performance yield with this approach may be better.

This will however significantly complexify the server.

Reply:
Yes, you are right. Works better with little count of queues, fixed too